docs/source/generate_agent_list.py

```diff
@@ -3,10 +3,83 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.abs
 
+import parlai.core.agents
 import parlai.agents
+from parlai.core.params import ParlaiParser
 
 import os
 import pkgutil
+import importlib
+import inspect
+import io
+import csv
+
+
+"""
+Extract the readme and CLI args for all of the standard agents.
+"""
+
+
+def _make_argparse_table(class_):
+    """
+    Build the reStructuredText table containing the args and descriptions.
+    """
+    readme = []
+    parser = ParlaiParser(False, False)
+    class_.add_cmdline_args(parser)
+    # group by whatever ArgumentGroups there are
+    for ag in parser._action_groups:
+        actions = []
+        # get options defined within only this group
+        for action in ag._group_actions:
+            if hasattr(action, 'hidden') and action.hidden:
+                # some options are marked hidden
+                continue
+            action_strings = ",  ".join(f'``{a}``' for a in action.option_strings)
+            description = []
+            # start with the help message
+            if action.help:
+                h = action.help
+                if not h[0].isupper():
+                    h = h[0].upper() + h[1:]
+                description += [h]
+            # list choices if there are any
+            if action.choices:
+                description += [
+                    "Choices: " + ", ".join(f'``{c}``' for c in action.choices) + "."
+                ]
+            # list default and recommended values.
+            default_value = ""
+            if action.default is not None:
+                default_value += f"Default: ``{action.default}``.  "
+            if hasattr(action, 'recommended') and action.recommended:
+                default_value += f"Recommended: ``{action.recommended}``. "
```

Comment:
no new line here?

Reply:
No, I wanted to keep them next to each other